### DISCUSSION TITLE
Figuring out jenkins build # from Game Version

### DISCUSSION THREAD

**Discussion by actual-nh** [CleverRaven/Cataclysm-DDA#47049] [Q&A] [answered]
Say I have a game version (in this case, 0.E-9136-gdc59dbd7db, from #47040). How do I figure out a jenkins build number (or another method of telling what's gone into it and what hasn't)? Thanks!

**Comment by anothersimulacrum** [comment, marked as answer]
The bit after the `g` in `gdc59dbd7db` is a commit hash.
You can also fetch the tags from the cleverraven repo and look for `cdda-jenkins-b#####`.

**Reply by actual-nh** [reply]
I got you on the first part (the g was messing searches up). I'm not clear on the second. (And what's the number after the 0.E-, BTW?)

**Reply by anothersimulacrum** [reply]
Every build for the last few thousand is published on this repo, with a tab of `cdda-jenkins-b$(BUILDNUMBER)`. You can fetch the tags from this repo, then see the build commits in git log or checkout to to them.

**Reply by actual-nh** [reply]
Well, yes, I know about the jenkins tags (why would I ask about going from a game version to them otherwise); sorry!

**Comment by actual-nh** [comment]
Ah. The number is how many commits have happened between the last tag available and that commit. For future reference:

- `git checkout master`
- `git fetch --tags upstream master`
- `git describe --tags` will get you the most recent jenkins tag in your local master's state
- `git describe --tags [part after the g]` for the most recent jenkins tag that's before the one you're wondering about.

**Reply by anothersimulacrum** [reply]
Oh sorry, I meant to say that but glanced over it.
